lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -29,6 +32,144 @@
+ *  2. Find the tag(s) you are looking for.
+ *  3. Request changes to the attributes in those tag(s).
+ *
+ * Example
+ * ```php
+ *     $processor = new WP_HTML_Tag_Processor( $html );
+ *     if ( $processor->next_tag( array( 'tag_name' => 'option' ) ) ) {
+ *         $processor->set_attribute( 'selected', true );
+ *     }
+ * ```
+ *
+ * ### Finding tags
+ *
+ * The `next_tag()` function moves the internal cursor through
+ * your input HTML document until it finds a tag meeting any of
+ * the supplied restrictions in the optional query argument. If
+ * no argument is provided then it will find the next HTML tag,
+ * regardless of what kind it is.
+ *
+ * If you want to _find whatever the next tag is_
+ * ```php
+ *     $processor->next_tag();
+ * ```
+ *
+ * | Goal                                                      | Query                                                                              |
+ * |-----------------------------------------------------------|------------------------------------------------------------------------------------|
+ * | Find any tag.                                             | `$processor->next_tag();`                                                          |
+ * | Find next image tag.                                      | `$processor->next_tag( array( 'tag_name' => 'img' ) );`                            |
+ * | Find next tag containing the `fullwidth` CSS class.       | `$process->next_tag( array( 'class_name' => 'fullwidth' ) );`                      |
+ * | Find next image tag containing the `fullwidth` CSS class. | `$process->next_tag( array( 'tag_name' => 'img', 'class_name' => 'fullwidth' ) );` |
+ *
+ * If a tag was found meeting your criteria then `next_tag()`
+ * will return `true` and you can proceed to modify it. If it
+ * returns `false`, however, it failed to find the tag and
+ * moved the cursor to the end of the file.
+ *
+ * Once the cursor reaches the end of the file the processor
+ * is done and if you want to reach an earlier tag you will
+ * need to recreate the processor and start over. The internal
+ * cursor can only proceed forward, never backing up.
+ *
+ * #### Custom queries
+ *
+ * Sometimes it's necessary to further inspect an HTML tag than
+ * the query syntax here permits. In these cases one may further
+ * inspect the search results using the read-only functions
+ * provided by the processor or external state or variables.
+ *
+ * Example
+ * ```php
+ *     // Paint up to the first five DIV or SPAN tags marked with the "jazzy" style.
+ *     $remaining_tags_to_style = 5;
+ *     while ( $processor->next_tag() ) {
+ *         if (
+ *              ( 'DIV' === $processor->get_tag() || 'SPAN' === $processor->get_tag() ) &&
+ *              'jazzy' === $processor->get_attribute( 'data-style' )
+ *         ) {
+ *             $processor->add_class( 'theme-style-everest-jazz' );
+ *             if ( 0 === --$remaining_tag_to_style ) {
```

Comment:
I missed the `--` the first time I read this, let's split it into two lines to make the intention clearer:

```suggestion
 *             --$remaining_tag_to_style;
 *             if ( 0 === $remaining_tag_to_style ) {
```

Reply:
resolved in be5015642e by moving the condition into the `while`